vagrant/scripts/perf-test.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -e
+
+# Exemplar script output
+#
+# Starting perf tests
+# Setting up avalanche...
+# Running 1 avalanche(s) with the following arguments:
+# --metric-count=200
+# --series-count=200
+# --port=9006
+# --series-interval=60000
+# --metric-interval=60000
+# --value-interval=30
+#
+# Letting prometheus ingest metrics for 20s...
+# Cleaning up avalanche...
+#
+# Prometheus container memory RSS:
+#         start:           657Mi
+#         finish:         1200Mi
+#         diff:            543Mi
+# receiver-mock ingested 3575 metrics during the test
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+AVALANCHE_YAML_PATH="${DIR}/../k8s/avalanche.yaml"
+readonly RECEIVER_MOCK_POD="$(kubectl get pod -n receiver-mock --no-headers -lapp=receiver-mock --output custom-columns=NAME:.metadata.name)"
+
+function is_avalanche_running() {
+    out="$(kubectl get pods --no-headers -n avalanche --output custom-columns=NAME:.metadata.name)"
+    if [[ -z "${out}" ]]; then
+        echo 1
+    else
+        echo 0
+    fi
+}
+
+function container_memory_rss() {
+    local container_name="${1}"
+
+    kubectl exec -n receiver-mock "${RECEIVER_MOCK_POD}" \
+        -- curl -s -g -XPOST collection-prometheus-oper-prometheus.sumologic.svc.cluster.local:9090/api/v1/query \
+        --data-urlencode "query=container_memory_rss{job='kubelet',container='${container_name}'}" \
+        | jq -r '.data.result[0].value[1]'
+}
+
+function ingested_metrics_count() {
+    local count
+    count="$(kubectl exec -n receiver-mock "${RECEIVER_MOCK_POD}" \
+        -- curl -s -XGET localhost:3000/metrics | grep "^receiver_mock_metrics_count" | awk '{print $2}')"
+    echo "${count}"
+}
+
+function reset_receiver_mock_metrics() {
+    kubectl exec -n receiver-mock "${RECEIVER_MOCK_POD}" \
+        -- curl -s -XPOST localhost:3000/metrics-reset >/dev/null
+}
+
+function run_test() {
+    local duration_seconds="${1}"
+    echo "Starting perf tests"
+
+    if [[ $(is_avalanche_running) -eq 0 ]]; then
+        echo "Avalanche is running, stop it, wait for the collection to cool down and then rerun the test"
+        exit 1
+    fi
+
+    PROMETHEUS_MEMORY_RSS_START=$(container_memory_rss prometheus)
+
+    echo "Setting up avalanche..."
+    kubectl apply --wait -f "${AVALANCHE_YAML_PATH}" >/dev/null || \
+        (echo "Failed applying avalanche yaml (${AVALANCHE_YAML_PATH})" && exit 1)
+    kubectl wait --timeout=120s -n avalanche --for=condition=Available deployment/avalanche >/dev/null
```

Comment:
Shouldn't we escape space between `Available` and `deployment`?

Reply:
Hmm no, why? even `kubectl wait --help` has a similar example:

```
Examples:
  # Wait for the pod "busybox1" to contain the status condition of type "Ready".
  kubectl wait --for=condition=Ready pod/busybox1
```

Reply:
:facepalm: thats fine